ledger-core/virtual/handlers/vdelegatedrequestfinished.go

```diff
@@ -149,32 +149,27 @@ func (s *SMVDelegatedRequestFinished) updateSharedState(
 
 	switch s.Payload.CallFlags.GetInterference() {
 	case contract.CallIntolerable:
-		if state.ActiveUnorderedPendingCount > 0 {
-			state.ActiveUnorderedPendingCount--
-		} else {
+		if state.PreviousExecutorUnorderedPendingCount == 0 {
 			ctx.Log().Warn(unexpectedVDelegateRequestFinished{
 				Object:  objectRef.String(),
 				Request: requestRef.String(),
 				Ordered: false,
 			})
 		}
 	case contract.CallTolerable:
-		if state.ActiveOrderedPendingCount > 0 {
-			state.ActiveOrderedPendingCount--
-
-			if state.ActiveOrderedPendingCount == 0 {
-				// If we do not have pending ordered, release sync object.
-				if !ctx.CallBargeIn(state.AwaitPendingOrdered) {
-					ctx.Log().Warn("AwaitPendingOrdered BargeIn receive false")
-				}
-			}
-		} else {
+		if state.PreviousExecutorOrderedPendingCount == 0 {
```

Comment:
this is not correct, this should check against PreviousExecutorOrderedPendingCount and size of the corresponding table.